### PR TITLE
improvement: S3C-2304 Update Dockerfile and README

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Update root Dockerfile and README.md.

### CODE REVIEW

**Review by JianqinWang — DISMISSED** [review]

**Review by JianqinWang — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,19 +1,16 @@
...
+    && apt-get install g++ -y \
     && npm cache clear --force \
-    && rm -rf ~/.node-gyp \
-    && rm -rf /tmp/npm-*
-
-# Keep the .git directory in order to properly report version
-COPY . /usr/src/app
+    && npm install --production
 
 ENTRYPOINT ["/usr/src/app/docker-entrypoint.sh"]
```
Comment (jonathan-gramain) [inline Dockerfile:12]: At this point, could be interesting to `apt-get remove` the build-only dependencies introduced like `g++` and `git` to reduce the image size.

**Review by JianqinWang — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (bennettbuchanan) [reply Dockerfile:12]: Great point! I'll try that out and update.

Comment (bennettbuchanan) [reply Dockerfile:12]: Updated to remove those dependencies.

File: Dockerfile [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
+FROM node:10-slim
 
 WORKDIR /usr/src/app
 
-COPY package.json /usr/src/app
+COPY . /usr/src/app
 
 RUN apt-get update \
-    && apt-get install -y jq --no-install-recommends \
+    && apt-get install git -y \
```
Comment (jonathan-gramain) [inline Dockerfile]: minor style (might also slightly increase speed of execution): you can simply concatenate all packages to install `apt-get install -y git python ...`, also why `apt install build-essential` not `apt-get`?

File: Dockerfile [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
+    && apt-get install g++ -y \
     && npm install --production \
-    && rm -rf /var/lib/apt/lists/* \
-    && npm cache clear --force \
-    && rm -rf ~/.node-gyp \
-    && rm -rf /tmp/npm-*
-
-# Keep the .git directory in order to properly report version
-COPY . /usr/src/app
+    && apt-get remove git -y \
```
Comment (jonathan-gramain) [inline Dockerfile]: same: `apt-get remove -y git g++`

**Review by jonathan-gramain — DISMISSED** [review]

**Review by JianqinWang — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Comment by bennettbuchanan** [comment]
@bert-e approve

**Pull request merged**